src/framebuffer.rs

```diff
@@ -178,18 +189,30 @@ const fn shift_msb0(input: usize) -> usize {
 }
 
 macro_rules! impl_bit {
-    (@impl $raw_storage_type:ty, $raw_type:ty, $shifter:expr) => {
+    (@impl $raw_storage_type:ty, $raw_type:ty, $shifter:expr, $dir:ty) => {
```

Comment:
Is it possible to use remove the `$dir` parameter from the macro and move it to a `DIR where DIR: PixelArrangement` type parameter, like in `ImageRaw`?  That remove some clutter from the docs, which are already pretty bad.